File: android/app/src/main/java/com/kryptos/android/core/StegoMode.kt
package com.kryptos.android.core

enum class StegoMode(val key: String) {
    WORDS("words"),
    SMART("smart"),
    LETTERS("letters");

    companion object {
        fun resolve(raw: String?, legacySmart: Boolean): StegoMode =
            entries.firstOrNull { it.key == raw } ?: if (legacySmart) SMART else WORDS
    }
}

File: CipherCore/Sources/CipherCore/Steganography/StegoMode.swift
import Foundation

public enum StegoMode: String, Sendable, CaseIterable, Identifiable {
    case words
    case smart
    case letters

    public var id: String { rawValue }

    public static func resolve(_ raw: String?, legacySmart: Bool) -> StegoMode {
        if let raw, let mode = StegoMode(rawValue: raw) { return mode }
        return legacySmart ? .smart : .words
    }
}
